src/rdymesh.c

```diff
@@ -878,6 +878,18 @@ static PetscErrorCode SaveNaturalCellIDs(DM dm, RDyCells *cells, PetscMPIInt ran
     PetscCall(VecDestroy(&natural));
     PetscCall(VecDestroy(&global));
     PetscCall(VecDestroy(&local));
+  } else {
+    PetscMPIInt commsize;
+    MPI_Comm    comm;
+    PetscCall(PetscObjectGetComm((PetscObject)dm, &comm));
+    PetscCallMPI(MPI_Comm_size(comm, &commsize));
```

Comment:
You know, we could do the same thing to find the `rank` used in line 846, instead of passing it in.

Reply:
Good point. Let me update the code accordingly.